slurmformspawner/slurm.py

```diff
@@ -37,6 +37,7 @@ def get_node_info(self):
                 output['cpu'].append(int(node['CPUTot']))
                 output['mem'].append(int(node['RealMemory']) - int(node.get('MemSpecLimit', '0')))
                 output['gres'].append(node['Gres'])
+                output['partitions'] = node['Partitions'].split(",")
```

Comment:
This line overwrites the partition list for each node. This is fine if all nodes belong to all partition, but otherwise, the partition list will only contain the partitions that contain the last node of the list of nodes.

On line 24, you need to add the key-pair `'partitions' : []` to the `output` dictionary, and for each node, extends the list with `node['Partitions'].split(",")`.